### PR TITLE
[RW-321][risk=low] Make flexible dependency versions fixed

### PR DESCRIPTION
**Pull request by rushtong**
## Addresses 
* Lock dependency versions: https://precisionmedicineinitiative.atlassian.net/browse/RW-321
* Upgrade okhttp version: https://precisionmedicineinitiative.atlassian.net/browse/RW-699

This PR sets all flexibly specified versions to the actual version that gets pulled in. While there, I addressed the okhttp story since it was easy.

This PR does not address many reported snyk errors. In the cases where there are fixes, they require code changes which is not in scope. Several others have no remediation. I am detailing those here for reference.

## Choosing not to upgrade
Upgrading the following library causes problems for some developers. Choosing to ignore this until we have a better solution. 
```
✗ Medium severity vulnerability found in com.google.guava:guava
  Description: Deserialization of Untrusted Data
  Info: https://snyk.io/vuln/SNYK-JAVA-COMGOOGLEGUAVA-32236
  Introduced through: com.google.cloud.sql:mysql-socket-factory@1.0.10, project :common-api@*, com.google.guava:guava@22.0, com.google.cloud:google-cloud-bigquery@0.30.0-beta, com.google.cloud:google-cloud-storage@1.8.0, com.google.truth:truth@0.42
  From: com.google.cloud.sql:mysql-socket-factory@1.0.10 > com.google.cloud.sql:jdbc-socket-factory-core@1.0.10 > com.google.guava:guava@22.0
  From: project :common-api@* > com.google.guava:guava@22.0
  From: com.google.guava:guava@22.0
  and 33 more...
```

## Out of scope
Any spring library upgrade causes compilation problems somewhere in the stack. 
```
✗ Low severity vulnerability found in org.springframework:spring-webmvc
  Description: Multipart Content Pollution
  Info: https://snyk.io/vuln/SNYK-JAVA-ORGSPRINGFRAMEWORK-32199
  Introduced through: org.springframework.boot:1.5.3.RELEASE@*, project :common-api@*
  From: org.springframework.boot:1.5.3.RELEASE@* > org.springframework:spring-webmvc@4.3.8.RELEASE
  From: project :common-api@* > org.springframework.boot:1.5.3.RELEASE@* > org.springframework:spring-webmvc@4.3.8.RELEASE

✗ Medium severity vulnerability found in org.springframework:spring-webmvc
  Description: Cross-Site Tracing (XST)
  Info: https://snyk.io/vuln/SNYK-JAVA-ORGSPRINGFRAMEWORK-31690
  Introduced through: org.springframework.boot:1.5.3.RELEASE@*, project :common-api@*
  From: org.springframework.boot:1.5.3.RELEASE@* > org.springframework:spring-webmvc@4.3.8.RELEASE
  From: project :common-api@* > org.springframework.boot:1.5.3.RELEASE@* > org.springframework:spring-webmvc@4.3.8.RELEASE

✗ Medium severity vulnerability found in org.springframework:spring-web
  Description: Authentication Bypass
  Info: https://snyk.io/vuln/SNYK-JAVA-ORGSPRINGFRAMEWORK-31644
  Introduced through: org.springframework.security:spring-security-web@4.2.2.RELEASE, org.springframework.boot:1.5.3.RELEASE@*, project :common-api@*
  From: org.springframework.security:spring-security-web@4.2.2.RELEASE > org.springframework:spring-web@4.3.8.RELEASE
  From: org.springframework.boot:1.5.3.RELEASE@* > org.springframework:spring-web@4.3.8.RELEASE
  From: org.springframework.boot:1.5.3.RELEASE@* > org.springframework:spring-webmvc@4.3.8.RELEASE > org.springframework:spring-web@4.3.8.RELEASE
  and 2 more...

✗ Medium severity vulnerability found in org.springframework:spring-web
  Description: Information Exposure
  Info: https://snyk.io/vuln/SNYK-JAVA-ORGSPRINGFRAMEWORK-31689
  Introduced through: org.springframework.security:spring-security-web@4.2.2.RELEASE, org.springframework.boot:1.5.3.RELEASE@*, project :common-api@*
  From: org.springframework.security:spring-security-web@4.2.2.RELEASE > org.springframework:spring-web@4.3.8.RELEASE
  From: org.springframework.boot:1.5.3.RELEASE@* > org.springframework:spring-web@4.3.8.RELEASE
  From: org.springframework.boot:1.5.3.RELEASE@* > org.springframework:spring-webmvc@4.3.8.RELEASE > org.springframework:spring-web@4.3.8.RELEASE
  and 2 more...

✗ High severity vulnerability found in org.springframework:spring-webmvc
  Description: Directory Traversal
  Info: https://snyk.io/vuln/SNYK-JAVA-ORGSPRINGFRAMEWORK-32202
  Introduced through: org.springframework.boot:1.5.3.RELEASE@*, project :common-api@*
  From: org.springframework.boot:1.5.3.RELEASE@* > org.springframework:spring-webmvc@4.3.8.RELEASE
  From: project :common-api@* > org.springframework.boot:1.5.3.RELEASE@* > org.springframework:spring-webmvc@4.3.8.RELEASE

✗ High severity vulnerability found in org.springframework.security:spring-security-core
  Description: Deserialization of Untrusted Data
  Info: https://snyk.io/vuln/SNYK-JAVA-ORGSPRINGFRAMEWORKSECURITY-31509
  Introduced through: org.springframework.security:spring-security-web@4.2.2.RELEASE
  From: org.springframework.security:spring-security-web@4.2.2.RELEASE > org.springframework.security:spring-security-core@4.2.2.RELEASE

✗ High severity vulnerability found in org.springframework.security:spring-security-core
  Description: Access Restriction Bypass
  Info: https://snyk.io/vuln/SNYK-JAVA-ORGSPRINGFRAMEWORKSECURITY-32123
  Introduced through: org.springframework.security:spring-security-web@4.2.2.RELEASE
  From: org.springframework.security:spring-security-web@4.2.2.RELEASE > org.springframework.security:spring-security-core@4.2.2.RELEASE

✗ High severity vulnerability found in org.springframework.data:spring-data-commons
  Description: Arbitrary Code Execution
  Info: https://snyk.io/vuln/SNYK-JAVA-ORGSPRINGFRAMEWORKDATA-32219
  Introduced through: org.springframework.boot:spring-boot-starter-data-jpa@1.5.3.RELEASE, project :common-api@*
  From: org.springframework.boot:spring-boot-starter-data-jpa@1.5.3.RELEASE > org.springframework.data:spring-data-jpa@1.11.3.RELEASE > org.springframework.data:spring-data-commons@1.13.3.RELEASE
  From: project :common-api@* > org.springframework.boot:spring-boot-starter-data-jpa@1.5.3.RELEASE > org.springframework.data:spring-data-jpa@1.11.3.RELEASE > org.springframework.data:spring-data-commons@1.13.3.RELEASE

✗ High severity vulnerability found in org.springframework.data:spring-data-commons
  Description: Denial of Service (DoS)
  Info: https://snyk.io/vuln/SNYK-JAVA-ORGSPRINGFRAMEWORKDATA-32231
  Introduced through: org.springframework.boot:spring-boot-starter-data-jpa@1.5.3.RELEASE, project :common-api@*
  From: org.springframework.boot:spring-boot-starter-data-jpa@1.5.3.RELEASE > org.springframework.data:spring-data-jpa@1.11.3.RELEASE > org.springframework.data:spring-data-commons@1.13.3.RELEASE
  From: project :common-api@* > org.springframework.boot:spring-boot-starter-data-jpa@1.5.3.RELEASE > org.springframework.data:spring-data-jpa@1.11.3.RELEASE > org.springframework.data:spring-data-commons@1.13.3.RELEASE
```

## No Remediation
```
✗ High severity vulnerability found in org.mortbay.jetty:jetty
  Description: Arbitrary Command Execution
  Info: https://snyk.io/vuln/SNYK-JAVA-ORGMORTBAYJETTY-32091
  Introduced through: com.google.oauth-client:google-oauth-client-jetty@1.23.0
  From: com.google.oauth-client:google-oauth-client-jetty@1.23.0 > org.mortbay.jetty:jetty@6.1.26

✗ High severity vulnerability found in com.h2database:h2
  Description: Arbitrary Code Execution
  Info: https://snyk.io/vuln/SNYK-JAVA-COMH2DATABASE-31685
  Introduced through: com.h2database:h2@1.4.194
  From: com.h2database:h2@1.4.194

✗ High severity vulnerability found in ch.qos.logback:logback-core
  Description: Arbitrary Code Execution
  Info: https://snyk.io/vuln/SNYK-JAVA-CHQOSLOGBACK-30208
  Introduced through: org.springframework.boot:1.5.3.RELEASE@*, project :common-api@*
  From: org.springframework.boot:1.5.3.RELEASE@* > org.springframework.boot:spring-boot-starter@1.5.3.RELEASE > org.springframework.boot:spring-boot-starter-logging@1.5.3.RELEASE > ch.qos.logback:logback-classic@1.1.11 > ch.qos.logback:logback-core@1.1.11
  From: project :common-api@* > org.springframework.boot:1.5.3.RELEASE@* > org.springframework.boot:spring-boot-starter@1.5.3.RELEASE > org.springframework.boot:spring-boot-starter-logging@1.5.3.RELEASE > ch.qos.logback:logback-classic@1.1.11 > ch.qos.logback:logback-core@1.1.11

✗ High severity vulnerability found in ch.qos.logback:logback-classic
  Description: Arbitrary Code Execution
  Info: https://snyk.io/vuln/SNYK-JAVA-CHQOSLOGBACK-31407
  Introduced through: org.springframework.boot:1.5.3.RELEASE@*, project :common-api@*
  From: org.springframework.boot:1.5.3.RELEASE@* > org.springframework.boot:spring-boot-starter@1.5.3.RELEASE > org.springframework.boot:spring-boot-starter-logging@1.5.3.RELEASE > ch.qos.logback:logback-classic@1.1.11
  From: project :common-api@* > org.springframework.boot:1.5.3.RELEASE@* > org.springframework.boot:spring-boot-starter@1.5.3.RELEASE > org.springframework.boot:spring-boot-starter-logging@1.5.3.RELEASE > ch.qos.logback:logback-classic@1.1.11

```

### CODE REVIEW

File: api/build.gradle [hunk 1]
```diff
@@ -239,38 +239,38 @@ dependencies {
...
   compile project(":common-api")
   compile 'com.google.api-client:google-api-client-appengine:1.23.0'
-  compile 'mysql:mysql-connector-java:+'
-  compile 'com.google.cloud.sql:mysql-socket-factory:+'
+  compile 'mysql:mysql-connector-java:8.0.11'
+  compile 'com.google.cloud.sql:mysql-socket-factory:1.0.10'
   compile "com.google.appengine:appengine:${GAE_VERSION}"
   compile "com.google.appengine:appengine-api-1.0-sdk:${GAE_VERSION}"
-  compile('org.springframework.boot:spring-boot-starter-web:+') {
+  compile('org.springframework.boot:spring-boot-starter-web') {
```
Comment (calbach) [inline api/build.gradle]: Hm, what does a lack of a version imply here?

Comment (rushtong) [reply api/build.gradle]: It gets picked up from the version specified in the plugin.

Comment (rushtong) [reply api/build.gradle]: To be extra clear, we can add it here too. Its just a little extraneous.

Comment (calbach) [reply api/build.gradle]: Oh cool, I guess this is what is being used here: https://stackoverflow.com/questions/41676534/how-are-some-gradle-dependencies-working-with-no-version-supplied

Yeah - this is nice, lets definitely leave it off where we can.

**Comment by rushtong** [comment]
@calbach - good catch, I missed that file completely. Will look at that one too.

File: api/build.gradle (hunk 1 above)
Comment (rushtong) [reply api/build.gradle]: Good find on the SO link. I didn't know the actual name of what's happening here - good to put a name to something I've only known through practice.

**Pull request merged**